csharp-ovh/Client/Client.Iterate.cs

```diff
@@ -0,0 +1,81 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Ovh.Api
+{
+    public partial class Client
+    {
+        /// <summary>
+        /// Issues a cascading GET call
+        /// </summary>
+        /// <param name="target">API method to call</param>
+        /// <param name="rootCallKwargs">Arguments to append to the parent URL</param>
+        /// <param name="childrenCallsKwargs">Arguments to append to the child URL</param>
+        /// <param name="childUrlFormat">Format of the url for subsequent calls. Use * as the placeholder. Defaults to "&lt;target&gt;/*</param>
+        /// <param name="needAuth">If true, send authentication headers</param>
+        /// <returns>An enumerator over children API Calls</returns>
+        public IEnumerable<string> Enumerate(string target, QueryStringParams rootCallKwargs = null,
+                                                 QueryStringParams childrenCallsKwargs = null,
+                                                 string childUrlFormat = null, bool needAuth = true)
+        {
+            if(childUrlFormat == null)
+            {
+                childUrlFormat = target + "/*";
+            }
+
+            int placeholderCount = childUrlFormat.Count(c => c == '*');
+            if(placeholderCount == 0)
+            {
+                throw new ArgumentException("Missing placeholder for childUrlFormat", "childUrlFormat");
+            }
+            else if(placeholderCount > 1)
+            {
+                throw new ArgumentException("Too many placeholders for childUrlFormat, only one is allowed", "childUrlFormat");
+            }
+
+            target += rootCallKwargs?.ToQueryString();
+            string childKwargs = childrenCallsKwargs?.ToQueryString();
+
+            foreach (string item in Call<IEnumerable<string>>("GET", target, null, needAuth))
+            {
+                string childUrl = getChildUrl(childUrlFormat, item, childKwargs);
+                yield return Call("GET", childUrl, null, needAuth, isBatch: false);
```

Comment:
Just curious : don't you have access to the `HttpMethod` enum here?

Reply:
Indeed, I could. But yeah that's just an enum so I would have to have a switch inside `Call` for that. Do you think this would read better by changing them all to the enum?